config/services.yaml

```diff
@@ -11,14 +11,26 @@ services:
         autowire: true      # Automatically injects dependencies in your services.
         autoconfigure: true # Automatically registers your services as commands, event subscribers, etc.
 
+    # Tags
+    _instanceof:
+        App\Processor\Converter\DataConverterInterface:
+            tags: ['app.data_converter']
+
     # makes classes in src/ available to be used as services
     # this creates a service per class whose id is the fully-qualified class name
     App\:
         resource: '../src/*'
         exclude: '../src/{DependencyInjection,Entity,Migrations,Tests,Kernel.php}'
 
-    Akeneo\PimEnterprise\ApiClient\AkeneoPimEnterpriseClientBuilder:
+    Box\Spout\Writer\CSV\Writer: ~
+
+    App\ApiClientFactory:
         arguments:
-            $baseUri: '%env(AKENEO_API_BASE_URI)%'
+            - '%env(AKENEO_API_BASE_URI)%'
```

Comment:
you should specify empty env variable I think:

```
parameters:
    env(API_URL): 'Your forgot to set API_URL environment variable.'
    env(API_USER): 'Your forgot to set API_USER environment variable.'
    env(API_PASSWORD): 'Your forgot to set API_PASSWORD environment variable.'
    env(API_CLIENT): 'Your forgot to set API_CLIENT environment variable.'
env(API_SECRET): 'Your forgot to set API_SECRET environment variable.'
```

Otherwise it prevents the cache to compile if it's not specified.

Also, you can avoid to create a new class by doing it this way:


```
    Akeneo\Pim\ApiClient\AkeneoPimClientBuilder:
        arguments:
            $baseUri: '%akeneo_api.url%'

    Akeneo\Pim\ApiClient\AkeneoPimClient:
        factory:  'Akeneo\Pim\ApiClient\AkeneoPimClientBuilder:buildAuthenticatedByPassword'
        arguments:
            - '%akeneo_api.credentials.client%'
            - '%akeneo_api.credentials.secret%'
            - '%akeneo_api.credentials.user%'
- '%akeneo_api.credentials.password%'
```